### PR TITLE
Merging in version 0.2.9 of mosdepth to our fork

### PR DESCRIPTION
**Pull request by BrianBrenner**
Merging in the main repo into our fork so that we can use htslib 1.10.

### CODE REVIEW

File: stats.nim [hunk 1]
```diff
@@ -55,10 +55,10 @@ proc countRead*(rs: var bamStats, rec: Record, chrom: string = "") =
   rs.readLengthHist.inc(rec.b.core.l_qseq)
 
   # This field is dubious as it depends on the order of traversing the bam file
-  rs.lastReadPos = rec.start
+  rs.lastReadPos = rec.start.int
```
Comment (BrianBrenner) [inline stats.nim:58]: These are the only changes I needed to make to get it to compile, there weren't other merge conflicts

**Comment by BrianBrenner** [comment]
@chmille4 Not sure, if you've worked with this fork before, but from what I can tell the merge didn't break anything. I tested the web sample upload with htslib 1.10 and the new mosdepth and it worked fine.

**Comment by BrianBrenner** [comment]
The build for this has never passed in the past so should be fine to merge

**Comment by chmille4** [comment]
@DoubleB123 Nice Work! I have not worked with this fork before. This was all Sam and a contractor.

Were you able to compare results with the previous version? Were they identical?

**Comment by BrianBrenner** [comment]
I just compared with the sample upload. Since we sample the data the results weren't exactly the same, but pretty close. I can try just running mosdepth without sampling to make sure the results are the same

**Comment by BrianBrenner** [comment]
I tested mosdepth with a bam file and can confirm the output was identical

**Pull request merged**